pymeasure/instruments/anaheimautomation/dpseriesmotorcontroller.py

```diff
@@ -0,0 +1,377 @@
+        2: Encoder Error 1: encoder needed to correct the motor position.
+        4: Encoder Error 2: encoder could not finish motor position correction.
+        8: Command Error: a bad command was sent to the controller.
+        16: Motor Error: motor speed profiles are set incorrectly.
+        32: Range Overflow Error: go to position has an overflow error.
+        64: Range Error: invalid number of commands and characters sent to the controller.
+        128: Transmit Error: Too many parameters sent back to the pc.
+        256: Mode Error: Controller is in a wrong mode.
+        512: Zero Parameters Error: Command sent to the controller that expected to see
+                                    parameters follow, but none were given.
+        1024: Busy Error: The controller is busy indexing (moving a motor).
+        2048: Memory Range Error: Specified address is out of range.
+        4096: Memory Command Error: Command pulled from memory is invalid.
+        8192: Thumbwheel Read Error: Error reading the thumbwheel, or thumbwheel is not present.
+    """
+    NO_ERR = 0
+    RCV_OVERFLOW_ERR = 1
+    ENC_ERR_1 = 2
+    ENC_ERR_2 = 4
+    CMD_ERR = 8
+    MOT_ERR = 16
+    RANGE_OVERFLOW_ERR = 32
+    RANGE_ERR = 64
+    TX_ERR = 128
+    MODE_ERR = 256
+    ZERO_PARAMS_ERR = 512
+    BUSY_ERR = 1024
+    MEM_RANGE_ERR = 2048
+    MEM_CMD_ERR = 4096
+    THBWHEEL_ERR = 8192
+
+    
+class DPSeriesMotorController(Instrument):
+    """Driver to interface with Anaheim Automation DP series stepper motor controllers. This driver has been tested with
+       the DPY50601 and DPE25601 motor controllers.
+    """
+
+    address = Instrument.control(
+        "%", "~%i",
+        """Integer property representing the address that the motor controller uses for serial communications.""",
+        validator=strict_range,
+        values=[0, 99],
+    )
+    
+    basespeed = Instrument.control(
+        "VB", "B%i", 
+        """Integer property that represents the motor controller's starting/homing speed. This property can be set.""",
+        validator=truncated_range,
+        values=[1, 5000],
+    )
+    
+    maxspeed = Instrument.control(
+        "VM", "M%i", 
+        """Integer property that represents the motor controller's maximum (running) speed. 
+           This property can be set.""",
+        validator=truncated_range,
+        values=[1, 50000],
+    )
+
+    direction = Instrument.control(
+        "V+", "%s",
+        """A string property that represents the direction in which the stepper motor will rotate upon subsequent 
+           step commands. This property can be set. 'CW' corresponds to clockwise rotation and 'CCW' corresponds to 
+           counter-clockwise rotation.""",
+        map_values=True, 
+        validator=strict_discrete_set,
+        values={"CW" : "+", "CCW" : "-"},
+        get_process=lambda d: "+" if d == 1.0 else "-",
+    )    
+
+    steps = Instrument.control(
```

Comment:
Maybe a better name to help disambiguate between the different variations of steps/step/position: `planned_steps`?